src/test.ts

```diff
@@ -0,0 +1,176 @@
+import os from 'os';
+import S3 from './S3';
+import axios from 'axios';
+import Redis from './Redis';
+import urlParser from 'url';
+import logger from './Logger';
+import { config } from './config';
+import MediaWiki from './MediaWiki';
+import Downloader from './Downloader';
+import fs, { readFileSync } from 'fs';
+import * as QueryStringParser from 'querystring';
+import { isValidEmail, ensureTrailingChar } from './util';
+
+function IsURL(url: string) {
```

Comment:
I'm quite sure we can do that with a standard module?

Reply:
@kelson42 I was earlier using standard module(valid-Uri) but these modules only check if URI is formatted properly or not they don't check if these URI make sense for example these will let pass URI like - https://en.wikip where as this URI doesn't make any sense. I don't know any other module or could find it any where if you know please let me know

Reply:
or is it fine to let these types of URI to pass?

Reply:
Maybe you could use the module you talk about and check that you get HTTP 200 OK with a simple Axios get request? Or maybe just even the Axios simple get request?